modules/clightning/module.cpp

```diff
@@ -235,7 +245,44 @@ std::string clightning_des_offer(const std::string_view input) {
         result << hex_encode(compressed, 33);
     }
     
-    clean_tmpctx();
+    return result.str();
+}
+
+std::optional<std::string> clightning_parse_p2p_lightning_message(std::span<const uint8_t> buffer) {
+    CleanTmpCtxGuard _cleanup;
+
+    u8 *msg = (u8 *) tal_arr(tmpctx, u8, buffer.size());
+    memcpy(msg, buffer.data(), buffer.size());
+    peer_wire msg_type = (enum peer_wire)fromwire_peektype(msg);
```

Comment:
Nit:  we should prefer C++-style casts for the extra compiler checks

```suggestion
    peer_wire msg_type = static_cast<enum peer_wire>(fromwire_peektype(msg));
```

Reply:
Updated!